plugin/execute_command.py

```diff
@@ -0,0 +1,56 @@
+import sublime
+from .core.registry import client_for_view, LspTextCommand
+from .core.settings import client_configs
+from .core.protocol import Request
+from .core.logging import debug
+from .core.rpc import Client
+
+try:
+    from typing import List, Optional, Dict, Any
+    assert List and Optional and Dict and Any
+except ImportError:
+    pass
+
+
+class LspExecuteCommand(LspTextCommand):
+    def __init__(self, view):
+        super().__init__(view)
+
+    def run(self, command_name) -> None:
+        client = client_for_view(self.view)
+        if not client:
+            self.view.show_popup("No connection with server")
+        else:
+            self._command_names = []   # type: List[str]
+            self._command_args = dict()   # type: Dict[str, Dict[str, Any]]
+            for config in client_configs.all:
+                for command in config.commands:
+                    self._command_names.append(command.name)
+                    self._command_args[command.name] = command.args
+
+            # TODO handle when command name is passed
+            if len(self._command_names) > 0:
+                self.view.window().show_quick_panel(self._command_names, lambda i: self._on_done(client, i))
```

Comment:
I still highly recommend using a `sublime_plugin.ListInputHandler` instead of `show_quick_panel`.

Reply:
no problem but since the run method returns None, how to link the `LspTextCommand` to to an instance of `ListInputHandler` ?

Reply:
> how to link the LspTextCommand to to an instance of ListInputHandler

You can define a new method for LspExecuteCommand:
```
    def input(self, args):
        return None if 'command_name' in args else CommandNameInputHandler(self.view)
```

where CommandNameInputHandler would be something like this:

```
class CommandNameInputHandler(sublime_plugin.ListInputHandler):
    def __init__(self, view):
        super(CommandNameInputHandler, self).__init__()
        self.view = view
        self.commands = []  # ... get a list of all the commands

    def list_items(self):
        return self.commands

    def placeholder(self):
        return self.commands[0]
```

Reply:
```
if type(command_name) is list:
                name = command_name[0]
                args = command_name[1] if len(command_name) == 2 else dict()
                self._send_command(client, name, args)
            elif type(command_name) is str:
                self._send_command(client, command_name, command_args)
```

This is the workaround that I found but had to removed type hints from the run method signature otherwise we get this from mypy
```
plugin/execute_command.py:48: error: Argument 3 to "_send_command" of "LspExecuteCommand" has incompatible type "Collection[str]"; expected "Dict[str, Any]"
```

not sure this is better than using plain `show_quick_panel`